ethcore/src/engines/clique/mod.rs

```diff
@@ -181,33 +178,38 @@ pub struct Clique {
 	pub block_state_by_hash: RwLock<LruCache<H256, CliqueBlockState>>,
 	pub proposals: RwLock<HashMap<Address, VoteType>>,
 	pub signer: RwLock<Option<Box<EngineSigner>>>,
-	pub step_service: Option<StepService>,
 }
 
 impl Clique {
 	/// Initialize Clique engine from empty state.
 	pub fn new(params: CliqueParams, machine: EthereumMachine) -> Result<Arc<Self>, Error> {
-		let mut engine = Clique {
+		/// Pause before starting to step Clique
+		const INITIAL_DELAY: Duration = Duration::from_secs(5);
+		/// Step Clique at most every 2 seconds
+		const SEALING_FREQ: Duration = Duration::from_secs(2);
+
+		let engine = Clique {
 			epoch_length: params.epoch,
 			period: params.period,
 			client: Default::default(),
 			block_state_by_hash: RwLock::new(LruCache::new(STATE_CACHE_NUM)),
 			proposals: Default::default(),
 			signer: Default::default(),
 			machine,
-			step_service: None,
 		};
-		if params.period > 0 {
-			engine.step_service = Some(StepService::new());
-			let engine = Arc::new(engine);
-			let weak_eng = Arc::downgrade(&engine);
-			if let Some(step_service) = &engine.step_service {
-				step_service.start(weak_eng);
-			}
-			Ok(engine)
-		} else {
-			Ok(Arc::new(engine))
-		}
+		let engine = Arc::new(engine);
+		let weak_eng = Arc::downgrade(&engine);
+
+		thread::Builder::new().name("StepService".into())
+			.spawn(move || {
+				thread::sleep(INITIAL_DELAY);
```

Comment:
Why is this delay needed? Seems smelly...

Reply:
Indeed, my guess is it's just to avoid triggering seals during startup to avoid some spam in the logs maybe? Seems safe to remove, but @niklasad1 knows better.

Reply:
Don't we have `engine.start()`? We could probably use this as a signal to that thread to start working.

Reply:
We do not have `engine.start()` (or I don't know where to look!)

Reply:
Oh, you're right. I thought that since we had `stop` we have `start` as well. We could either introduce it or maybe trigger `start()` on first `generate_seal`? I believe that function should be called for every engine by the miner module, no?

Reply:
Or we could use `register_client` as the trigger?

Reply:
I went with removing the delay altogether. IMO it muddles up the code (why 5sec? is it important or just an optimization? etc) for little gain. @niklasad1 Thoughts on this? :)